### DISCUSSION TITLE
if data exist than update the record

### DISCUSSION THREAD

**Discussion by breakingbad45** [pocketbase/pocketbase#5849] [Q&A]
![Screenshot (631)](https://github.com/user-attachments/assets/90a5d420-8ea8-414f-a35e-8d04219d8705)

if data exist than update the record is it possible?

**Comment by starredev** [comment]
You mean upsert? Its added in V0.23 but only for transactions i believe. 
Also you could easily create  by yourself as function in pb_hooks

**Reply by breakingbad45** [reply]
yes..can you explain how?

**Comment by chnfyi** [comment]
With Web API :

```javascript
const record = await pb.collection("your_collection").getFirstListItem('catalog="value"');
await pb.collection("your_collection").update(record.id, {
    title: 'Lorem ipsum',
});
``` 

With JSVM (v0.22.23) :
```javascript
const record = $app.dao().findFirstRecordByData("your_collection", "catalog", "value")
record.set("title", "Lorem ipsum")
$app.dao().saveRecord(record)
``` 

With JSVM (v0.23.0rc) :
```javascript
const record = $app.findFirstRecordByData("your_collection", "catalog", "value")
record.set("title", "Lorem ipsum")
$app.save(record)
``` 

If you want to do it through Web API, make sure to set your collection API rules according to your requirements. Otherwise you won't be able to retrieve or update records.
